File: Server/TravelService/Models/Activity.cs
namespace TravelService.Models
{
    public class Activity
    {
        public Guid Id { get; set; }
        public Guid TravelPlanId { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public string Time { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal EstimatedCost { get; set; }
        public string Status { get; set; } = "Planned";

        public TravelPlan TravelPlan { get; set; } = null!;
    }
}

File: client/src/features/activity/types/Activity.ts
import type { ActivityStatus } from "./ActivityStatus";

export interface Activity {
    id: string;
    travelPlanId: string;
    name: string;
    date: string;
    time: string;
    location: string;
    description: string;
    estimatedCost: number;
    status: ActivityStatus;
}